Add Liza's new spike protection docs [merge after feature launch]

shanamatthews opened the pull request:
<!-- Use this checklist to make sure your PR is ready for merge. You may delete any sections you don't need. -->

## Pre-merge checklist

*If you work at Sentry, you're able to merge your own PR without review, but please don't unless there's a good reason.*

- [ ] Checked Vercel preview for correctness, including links
- [ ] PR was reviewed and approved by any necessary SMEs
- [ ] PR was reviewed and approved by a member of the [Sentry docs team](https://github.com/orgs/getsentry/teams/docs)

## Description of changes

*Describe your changes here. If your PR relates to or resolves an issue, add a link to that too.*

## Legal Boilerplate

<!-- Sentry employees and contractors can delete or ignore this section. -->

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

## Extra resources

- [Sentry Docs contributor guide](https://docs.sentry.io/contributing/)

Dhrumil-Sentry commented:
Any reason why this link url is explicit compared to others?
![image](https://user-images.githubusercontent.com/109983869/229902328-ffc5477d-bf7d-46d4-93a4-b314ea34bc52.png)


Wrt to the following line 

> To review the error events that were dropped to save your quotaThe monthly number of events that you pay Sentry to track. with Spike Protection, go to the "Usage Stats" tab of the Stats page for your Sentry org and select "Errors" in the "Category" dropdown.

I would suggested adding the following:

> You can see project-level details of your stats by using the project selector. Sentry can show the spike threshold for a time duration greater than 6hrs and less than 30days
![image](https://user-images.githubusercontent.com/109983869/229903408-b97ce315-9470-46e6-ab5c-7c2a83ea96df.png)

Dhrumil-Sentry commented:
We still need to add some more content to the notifications section, just waiting for my dev team have this done

shanamatthews commented:
> Any reason why this link url is explicit compared to others? ![image](https://user-images.githubusercontent.com/109983869/229902328-ffc5477d-bf7d-46d4-93a4-b314ea34bc52.png)
> 
> Wrt to the following line
> 
> > To review the error events that were dropped to save your quotaThe monthly number of events that you pay Sentry to track. with Spike Protection, go to the "Usage Stats" tab of the Stats page for your Sentry org and select "Errors" in the "Category" dropdown.
> 
> I would suggested adding the following:
> 
> > You can see project-level details of your stats by using the project selector. Sentry can show the spike threshold for a time duration greater than 6hrs and less than 30days
> > ![image](https://user-images.githubusercontent.com/109983869/229903408-b97ce315-9470-46e6-ab5c-7c2a83ea96df.png)

I believe that link issue was just a typo! I've fixed it. I've also added in the new sentence & screenshot. 

@Dhrumil-Sentry - let me know when you have more on the notifications :D
